python/lsst/daf/butler/_butler_collections.py

```diff
@@ -27,15 +27,59 @@
 
 from __future__ import annotations
 
-__all__ = ("ButlerCollections",)
+__all__ = ("ButlerCollections", "CollectionInfo")
 
 from abc import ABC, abstractmethod
-from collections.abc import Iterable
+from collections.abc import Iterable, Sequence, Set
+from typing import Any, overload
 
+from pydantic import BaseModel
 
-class ButlerCollections(ABC):
+from ._collection_type import CollectionType
+
+
+class CollectionInfo(BaseModel):
+    """Information about a single Butler collection."""
+
+    name: str
+    """Name of the collection."""
+    type: CollectionType
+    """Type of the collection."""
+    doc: str = ""
+    """Documentation string associated with this collection."""
+    children: tuple[str, ...] = tuple()
+    """Children of this collection (only if CHAINED)."""
+    parents: frozenset[str] = frozenset()
+    """Any parents of this collection."""
+
+
+class ButlerCollections(ABC, Sequence):
     """Methods for working with collections stored in the Butler."""
 
+    @overload
+    def __getitem__(self, index: int) -> str: ...
+
+    @overload
+    def __getitem__(self, index: slice) -> Sequence[str]: ...
+
+    def __getitem__(self, index: int | slice) -> str | Sequence[str]:
+        return self.defaults[index]
+
+    def __len__(self) -> int:
+        return len(self.defaults)
+
+    def __eq__(self, other: Any) -> bool:
```

Comment:
This is only here because we have some tests that compare collections directly. It's probably best to remove this method and change the tests to compare `butler.collections.defaults` directly.